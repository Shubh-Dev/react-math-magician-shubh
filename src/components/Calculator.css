* {
  text-decoration: none;
  list-style-type: none;
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  background: linear-gradient(to right, #eea486, #dfd9e1);
  font: 20px/1.5 Arial, Helvetica, sans-serif;
}

.container {
  margin: 10vh auto 10vh auto;
  display: grid;
  grid-template-columns: repeat(4, 8rem);
  grid-template-rows: minmax(5rem, auto) repeat(5, 6rem);
  justify-content: center;
}

.bigger-btn {
  grid-column: span 2;
}

.display-area {
  grid-column: 1 / -1;
  background-color: rgb(138, 138, 134);
  color: #fff;
  font-size: 30px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 26px 5px 0 3px;
}

.container > button {
  cursor: pointer;
  font-size: 2rem;
  border: 1px solid #fff;
  outline: none;
}

.orange-color {
  background-color: orange;
}

.title-cal {
  margin-left: 20px;
}
