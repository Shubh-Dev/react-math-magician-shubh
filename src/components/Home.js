import React from 'react';

const Home = () => (
  <h1>Router Test</h1>
);

export default Home;
