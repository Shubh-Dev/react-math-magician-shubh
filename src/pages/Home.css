.homepage {
    width: 80%;
    margin: auto;
    padding-top: 13vh;
    color: rgb(56, 54, 46);
    font-size: 25px;
  }