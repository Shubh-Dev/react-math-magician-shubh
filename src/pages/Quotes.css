.quotesSection {
  width: 80%;
  margin: 13vh auto auto auto;
  font-size: 30px;
  text-align: center;
  font-weight: 600;
  color: rgb(72, 71, 68);
}
