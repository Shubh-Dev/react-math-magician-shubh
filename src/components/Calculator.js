import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/Calculate';

const Calculator = () => {
  const [result, setResult] = useState({ total: 0, next: null, operation: null });
  const handleClick = (e) => {
    setResult(calculate(result, e.target.value));
  };
  const { total, next, operation } = result;
  return (
    <>
      <h2 className="title-cal">Let&apos;s do some Math!!</h2>
      <div className="container">
        <div className="display-area">
          {total}

          {operation}

          {next}

        </div>
        <button type="button" value="AC" onClick={handleClick}>
          AC
        </button>
        <button type="button" value="+/-" onClick={handleClick}>
          +/-
        </button>
        <button type="button" value="%" onClick={handleClick}>
          %
        </button>
        <button type="button" className="orange-color" value="รท" onClick={handleClick}>
          รท
        </button>
        <button type="button" value="7" onClick={handleClick}>
          7
        </button>
        <button type="button" value="8" onClick={handleClick}>
          8
        </button>
        <button type="button" value="9" onClick={handleClick}>
          9
        </button>
        <button type="button" className="orange-color" value="x" onClick={handleClick}>
          x
        </button>
        <button type="button" value="4" onClick={handleClick}>
          4
        </button>
        <button type="button" value="5" onClick={handleClick}>
          5
        </button>
        <button type="button" value="6" onClick={handleClick}>
          6
        </button>
        <button type="button" className="orange-color" value="-" onClick={handleClick}>
          -
        </button>
        <button type="button" value="1" onClick={handleClick}>
          1
        </button>
        <button type="button" value="2" onClick={handleClick}>
          2
        </button>
        <button type="button" value="3" onClick={handleClick}>
          3
        </button>
        <button type="button" className="orange-color" value="+" onClick={handleClick}>
          +
        </button>
        <button type="button" className="bigger-btn" value="0" onClick={handleClick}>
          0
        </button>
        <button type="button" value="." onClick={handleClick}>
          .
        </button>
        <button type="button" className="orange-color" value="=" onClick={handleClick}>
          =
        </button>
      </div>
    </>
  );
};
export default Calculator;
