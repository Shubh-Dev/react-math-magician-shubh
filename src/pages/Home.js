import React from 'react';
import './Home.css';

const Homepage = () => (
  <div className="homepage">
    <h2>Welcome to our page!</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit,
      sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud
      exercitation ullamco laboris nisi ut aliquip ex ea
      commodo consequat. Duis aute irure dolor in reprehenderit
      in voluptate velit esse cillum dolore eu fugiat nulla
      pariatur. Excepteur sint occaecat cupidatat non proident,
      sunt in culpa qui officia deserunt mollit anim id est
      laborum.

      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing
        elit, sed do eiusmod tempor incididunt ut labore et
        dolore magna aliqua. Ut enim ad minim veniam, quis
        nostrud exercitation ullamco laboris nisi ut aliquip
        ex ea commodo consequat. Duis aute irure dolor in
        reprehenderit in voluptate velit esse cillum dolore
        eu fugiat nulla pariatur. Excepteur sint occaecat.
      </p>
    </p>
  </div>
);

export default Homepage;