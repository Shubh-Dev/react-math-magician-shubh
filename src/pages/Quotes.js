import React from 'react';
import './Quotes.css';

const Quotes = () => (
  <div className="quotesSection">
    <p>
      Mathematics is not about numbers, equation, computations or algorithms:
      it is about understanding.- Willium Paul Thurston.
    </p>
  </div>
);

export default Quotes;
