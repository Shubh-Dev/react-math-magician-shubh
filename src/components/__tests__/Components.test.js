import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../pages/Home';
import Quotes from '../../pages/Quotes';
import Calculator from '../Calculator';

describe('Calculater component test', () => {
  test('builds the snapshot of the Calculator component', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('builds the snapshot of the Home component', () => {
    const tree = renderer.create(<Home />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('builds the snapshot of the Quotes component', () => {
    const tree = renderer.create(<Quotes />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
