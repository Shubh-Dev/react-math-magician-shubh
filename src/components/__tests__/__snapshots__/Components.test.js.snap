// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculater component test builds the snapshot of the Calculator component 1`] = `
Array [
  <h2
    className="title-cal"
  >
    Let's do some Math!
  </h2>,
  <div
    className="container"
  >
    <div
      className="display-area"
    >
      0
    </div>
    <button
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
      value="x"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="bigger-btn"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>,
]
`;

exports[`Calculater component test builds the snapshot of the Home component 1`] = `
<div
  className="homepage"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.
    </p>
  </p>
</div>
`;

exports[`Calculater component test builds the snapshot of the Quotes component 1`] = `
<div
  className="quotesSection"
>
  <p>
    Mathematics is not about numbers, equation, computations or algorithms: it is about understanding.- Willium Paul Thurston.
  </p>
</div>
`;
